### PR TITLE
[PLAT-76309]Add doc guide to create a gcp private service connect workspace.

### PR DESCRIPTION
**Pull request by jessiedu-db**
[PLAT-76309]Add doc guide to create a gcp private service connect workspace.

[PLAT-76309]: https://databricks.atlassian.net/browse/PLAT-76309?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/guides/gcp-private-service-connect-workspace.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+data "google_client_config" "current" {
+}
+
+resource "random_string" "suffix" {
+  special = false
+  upper   = false
+  length  = 6
+}
+```
```
Comment (grayluck) [inline docs/guides/gcp-private-service-connect-workspace.md:147]: This block seems to be irrelevant to GCP PSC. Can we refer the https://src.dev.databricks.com/databricks/terraform-provider-databricks/-/blob/docs/guides/gcp-workspace.md#creating-a-gcp-service-account-for-databricks-provisioning pages instead of copy/paste?

File: docs/guides/gcp-private-service-connect-workspace.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+## Creating a VPC network
+
+The very first step is VPC creation with necessary resources. Please consult [main documentation page](https://docs.gcp.databricks.com/administration-guide/cloud-configurations/gcp/customer-managed-vpc.html) for **the most complete and up-to-date details on networking**. A GCP VPC is registered as [databricks_mws_networks](../resources/mws_networks.md) resource.
+
+To enable [back-end Private Service Connect (data plane to control plane)](https://docs.gcp.databricks.com/administration-guide/cloud-configurations/gcp/private-service-connect.html#two-private-service-connect-options), configure network with the two back-end VPC endpoints:
+- Back-end VPC endpoint for SSC relay
```
Comment (grayluck) [inline docs/guides/gcp-private-service-connect-workspace.md:154]: SCC is not mentioned on the doc. Can we add a link for details?

File: docs/guides/gcp-private-service-connect-workspace.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+In [the next step](workspace-management.md), please use the following configuration for the provider:
+
+```hcl
+provider "databricks" {
+  host  = module.dbx_gcp.workspace_url
+  token = module.dbx_gcp.token_value
+}
+```
+
+We assume that you have a terraform module in your project that creats a workspace (using [Databricks Workspace](#creating-a-databricks-workspace) section) and you named it as `dbx_gcp` while calling it in the **main.tf** file of your terraform project. And `workspace_url` and `token_value` are the output attributes of that module. This provider configuration will allow you to use the generated token during workspace creation to authenticate to the created workspace.
```
Comment (grayluck) [inline docs/guides/gcp-private-service-connect-workspace.md:311]: Unrelated to PSC. Do we need them?

**Review by grayluck — CHANGES_REQUESTED** [review]

File: docs/guides/gcp-private-service-connect-workspace.md (hunk 1 above)
Comment (jessiedu-db) [reply docs/guides/gcp-private-service-connect-workspace.md]: Done.

File: docs/guides/gcp-private-service-connect-workspace.md (hunk 2 above)
Comment (jessiedu-db) [reply docs/guides/gcp-private-service-connect-workspace.md]: Done.

File: docs/guides/gcp-private-service-connect-workspace.md (hunk 3 above)
Comment (jessiedu-db) [reply docs/guides/gcp-private-service-connect-workspace.md]: I copied it from gcp_workspace.md. I don't know what they are. Removed.
